src/shared/sidepanel/sidepanel-commands.background.ts

```diff
@@ -0,0 +1,56 @@
+import browser from 'webextension-polyfill';
+import { getSidepanelUrl } from '../getPopupUrl';
+import { closeSidepanel } from './sidepanel-messaging.background';
+import { browserState } from './BrowserState';
+import { isSidepanelSupported } from './sidepanel-support';
+import { getActiveTab } from './sidepanel-apis';
+
+export function initializeSidepanelCommands() {
+  if (!isSidepanelSupported()) {
+    return;
+  }
+  browser.commands.onCommand.addListener(async (command) => {
+    if (command === 'sidepanel_open_custom') {
+      const { currentWindowId } = browserState;
+      try {
+        /**
+         * What we try to do here is close sidepanel if it's open and open it if it's closed.
+         * But because we are only allowed to open it synchronously, we cannot do an async call
+         * to check if it is currently open or not.
+         * So the trick is to always call "close" first: if sidepanel is closed, there's noone to receive
+         * the request, so it's ok. But if it is open, it will receive the request asyncronously, and therefore
+         * it will react to it AFTER we call the {sidePanel.open} method later, so the {open} method will
+         * not work, and that's what we need.
+         */
+        closeSidepanel({ windowId: currentWindowId ?? null });
+      } catch {
+        // it's ok, it might've not been open at all
+      }
+      const url = getSidepanelUrl();
+      if (!currentWindowId) {
```

Comment:
Should we move this condition to the beginning of this function? To avoid closing sidepanel without opening it next

Reply:
Hm, I'm thinking that if user pressed a shortcut and sidepanel is opened and it's possible to close it, it should be closed. That way the user interaction won't seem broken